addons/purchase/migrations/9.0.1.2/end-migration.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Tecnativa - Jairo Llopis
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.logging()
+def deferred_pol_qty_received_compute(env):
+    """Complete migration of purchase.order.line#qty_received
+
+    Skip records that were optimized in SQL in post-migration.
+    """
+    # Execute https://bit.ly/2PgUKaT for matching records
+    pol_todo = env["purchase.order.line"].search([
```

Comment:
This search is not taking into account BoMs, so it's basically recomputing almost everything.

Reply:
Yes, I'm just moving the thing to the end, where mrp will be loaded already; I'm not actually optimizing this part.

Reply:
Last test migration took 13 min. here.

IMHO it's not worth the effort to optimize this part more.